src/features/transactionTable/components/columns/DeleteButton.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from '@/components/ui/alert-dialog';
+import { Button } from '@/components/ui/button';
+import { Trash as TrashIcon } from 'lucide-react';
+import { TransactionTableData } from '@/components/AddTransactionForm/AddTransactionForm';
+import { toast } from '@/hooks/useToast';
+import { useTransactionStore } from '@/stores/TransactionStore';
+
+type DeleteButtonProps = {
+  transactionToEdit: TransactionTableData;
+};
```

Comment:
1. There are just on component definition in the file, the typing can be defined just like `Props`. 
2. The component only needs the id transaction, there is no need to send the complete object.

```suggestion
type Props = {
  transactionId: TransactionTableData["id"];
};
```